doc/partitioned-db.md

```diff
@@ -0,0 +1,206 @@
+# Partitioned Database
+
+When creating a Cloudant database, you can configure it as non-partitioned or partitioned.
+A partitioned database offers significant performance and cost advantages but requires you to specify a logical partitioning of your data.
+
+Reference:
+
+- The driver module `nodejs-cloudant`: https://github.com/cloudant/nodejs-cloudant#partitioned-databases
+- The Cloudant official document: https://cloud.ibm.com/docs/services/Cloudant/guides?topic=cloudant-database-partitioning
+
+
+Let's start with a user scenario to understand what's been added in the new feature.
+
+_To use the following partition features, please make sure the Cloudant database that you connects to is created as partitioned._
+
+Suppose you have a Product model, with properties `id`, `name`, `tag`, `city`, and specify `city` as your partition key. You can define your model as
+
+```js
+Product = db.define('Product', {
+  name: {type: String},
+  tag: {type: String},
+  // specify `city` as the partitionKey
+  city: {type: String, isPartitionKey: true},
+  }, {
+    // include `id` field when creating a new product
+    forceId: false,
+    indexes: {
+      // create a partitioned index for frequently queried
+      // fields like `name`
+      'product_name_index': {
+        partitioned: true,
+        keys: {
+            name: 1
+        },
+      },
+    }
+  });
+```
+
+When creating a new product, you can generate its `id` field as `${city_name}: ${uuid()}`, and perform partitioned query by placing `city` in the query like 
+
+```js
+const filter = {
+  where: {
+    city: 'toronto',
+    name: 'food'
+  }
+}
+Product.find(filter);
+```
+
+OR provide it in query options like:
+
+```js
+const filter = {
+  where: {
+    name: 'food'
+  }
+};
+const options = {
+  partitionKey: 'toronto'
+}
+Product.find(filter, options);
+```
+
+Other than leveraging partitioned find, you can have a secondary query optimization by creating partitioned index for the frequently queried fields like:
+
+```js
+  // ...other model configuration fields
+  indexes: {
+    'product_name_index': {
+      partitioned: true,
+      keys: {
+          name: 1
+      },
+    },
+  }
+```
+
+As you've read, we've enabled:
+
+- Adding partitioned index
+- Creating document in Partitioned Database
+- Defining partitioned field
+- Performing partitioned find
+
+in this epic, next you can ready more about the details of each functionality.
+
+## Updating Driver
+
+Make sure you are above `loopback-connector-cloudant@2.4.0` to support partition features, and if you have `@cloudant/cloudant` installed, please upgrade it to version 4.x.
```

Comment:
I'd suggest to move this right under the section title `Partitioned Database`. 
Suggested change:
```Make sure you are using at least loopback-connector-cloudant@2.4.0 and @cloudant/cloudant@4.x```